miner/src/create_block_template/mod.rs

```diff
@@ -362,7 +387,9 @@ where
                     );
                     let mut blue_blocks = vec![];
 
-                    let __selected_parent = blues.remove(0);
+                    let selected_parent = blues.remove(0);
+                    previous_header =
+                        self.get_dag_previous_header(previous_header, selected_parent)?;
```

Comment:
What's the purpose to of updating previous_header here? Do we need to update the `miner`'s block forging process?

Reply:
I think maybe he wants to solve header not update but blockdag tips update. @jackzhhuang

Reply:
Yes

Reply:
I have two questions,  the second one？ Is there a issue tracking on it?

Reply:
It is necessary to update the chain object in miner service.
This is the case caused by the 0 dag fork height, see test_block_chain_reset.